sdks/go/pkg/beam/runners/universal/runnerlib/stage.go

```diff
@@ -17,27 +17,127 @@ package runnerlib
 
 import (
 	"context"
+	"io"
+	"os"
 	"time"
 
 	"github.com/apache/beam/sdks/go/pkg/beam/artifact"
+	"github.com/apache/beam/sdks/go/pkg/beam/core/runtime/graphx"
 	"github.com/apache/beam/sdks/go/pkg/beam/internal/errors"
 	jobpb "github.com/apache/beam/sdks/go/pkg/beam/model/jobmanagement_v1"
 	"github.com/apache/beam/sdks/go/pkg/beam/util/grpcx"
+	"google.golang.org/grpc"
 )
 
 // Stage stages the worker binary and any additional files to the given
 // artifact staging endpoint. It returns the retrieval token if successful.
-func Stage(ctx context.Context, id, endpoint, binary, st string, files ...artifact.KeyedFile) (retrievalToken string, err error) {
+func Stage(ctx context.Context, id, endpoint, binary, st string) (retrievalToken string, err error) {
 	ctx = grpcx.WriteWorkerID(ctx, id)
 	cc, err := grpcx.Dial(ctx, endpoint, 2*time.Minute)
 	if err != nil {
 		return "", errors.WithContext(err, "connecting to artifact service")
 	}
 	defer cc.Close()
 
+	err = StageViaPortableApi(ctx, cc, binary, st)
+
+	if err == nil {
+		return "", err
+	} else {
```

Comment:
Style nit: Idiomatically, if the "if" clause returns, elide the else block. Since Go has explicit brackets for if scopes, we avoid the "1 line if" in C(++) and accidental mis-alignment/scope errors with python.

I'm not 100% but this advice only applies to if-else blocks where the remainder of the code is in the else, not for example, in default:switch cases, where dropping out of the switch instead of default is aesthetically awkward.

Reply:
To me it depends on whether it is "one of two+ options" (in which case an explicit else clause is clearer) or "exit early vs. fallback" (in which case not having the else is clearer). Removing the else clause for now; hopefully soon we'll be able to simply remove the fallback altogether.